saiph/inverse_transform_test.py

```diff
@@ -117,24 +118,108 @@ def test_inverse_transform_deterministic() -> None:
+    )
+    # assert equal for the min as there are many low values
+    assert_series_equal(wbcd_statistics.loc["min"], reversed_statistics.loc["min"])
+    assert_allclose(wbcd_statistics.loc["25%"], reversed_statistics.loc["25%"], atol=1)
+    assert_allclose(wbcd_statistics.loc["50%"], reversed_statistics.loc["50%"], atol=1)
+    assert_allclose(wbcd_statistics.loc["75%"], reversed_statistics.loc["75%"], atol=2)
+    assert_series_equal(wbcd_statistics.loc["max"], reversed_statistics.loc["max"])
+
+
+def test_inverse_from_coord_pca(
+    wbcd_quanti_df: pd.DataFrame,
+    wbcd_supplemental_coord_quanti: pd.DataFrame,
+) -> None:
+    """Check that inverse supplemental coordinates using PCA yield correct results.
+
+    We use `use_max_modalities=False` to keep the data logical.
+    We compare indicators of the distributions for each column.
+    """
+    model = fit(wbcd_quanti_df, nf="all")
+    # wbcd_supplemental_df was obtained using
+    # inverse_transform with wbcd_supplemental_coord_quanti and seed=123
+    reversed_individuals = inverse_transform(
+        wbcd_supplemental_coord_quanti, model, use_max_modalities=False
+    )
+
+    reversed_statistics = reversed_individuals.describe()
+    wbcd_statistics = wbcd_quanti_df.describe()
+
+    assert_series_equal(wbcd_statistics.loc["count"], reversed_statistics.loc["count"])
+    assert_allclose(
+        wbcd_statistics.loc["mean"], reversed_statistics.loc["mean"], atol=0.4
+    )
+    assert_allclose(
+        wbcd_statistics.loc["std"], reversed_statistics.loc["std"], atol=0.7
+    )
+    # assert equal for the min as there are many low values
+    assert_series_equal(wbcd_statistics.loc["min"], reversed_statistics.loc["min"])
+    assert_allclose(wbcd_statistics.loc["25%"], reversed_statistics.loc["25%"], atol=1)
+    assert_allclose(wbcd_statistics.loc["50%"], reversed_statistics.loc["50%"], atol=1)
+    assert_allclose(wbcd_statistics.loc["75%"], reversed_statistics.loc["75%"], atol=1)
+    assert_series_equal(
+        wbcd_statistics.loc["max"], reversed_statistics.loc["max"], atol=1
+    )
+
+
+def test_inverse_from_coord_famd(
+    wbcd_mixed_df: pd.DataFrame,
+    wbcd_supplemental_coord_mixed: pd.DataFrame,
+) -> None:
+    """Check that inverse supplemental coordinates using FAMD yield correct results.
+
+    We use `use_max_modalities=False` to keep the data logical.
+    We compare indicators of the distributions for each column.
+    """
+    model = fit(wbcd_mixed_df, nf="all")
+    # wbcd_supplemental_df was obtained using
+    # inverse_transform with wbcd_supplemental_coord_quanti and seed=123
+    reversed_individuals = inverse_transform(
+        wbcd_supplemental_coord_mixed, model, use_max_modalities=False
+    )
+
+    reversed_statistics = reversed_individuals.describe()
+    wbcd_statistics = wbcd_mixed_df.describe()
+
+    assert_series_equal(wbcd_statistics.loc["count"], reversed_statistics.loc["count"])
```

Comment:
I am still wondering if this belongs to saiph or lsg. This test is almost like comparing if avatar statistics are similar to original statistics (with the inverse transform step). 

I write my thoughts here:
with transform_then_inverse tests (in the projection_test file) we already check if same coordinates are leading to same individuals.
Here we want to check if new coordinates (never seen with the individuals we have) lead to relevant individuals when inverse transformed.
Instead of providing avatars, we might want to make sure that supplemental individual having coordinates between two original individuals will still be between these two after inverse transform